src/commands/deployImage/deployImage.ts

```diff
@@ -95,7 +96,7 @@ export async function deployImage(context: ITreeItemPickerContext & Partial<IDep
             void showContainerAppCreated(updatedContainerApp, true);
         });
 
-        refreshContainerApp(containerApp.id);
+        refreshContainerAppEnvironment(containerApp.managedEnvironmentId);
```

Comment:
I think you can just change this to 

```ts
ext.state.notifyChildrenChanged(containerApp.managedEnvironmentId);
```

And avoid making all the other changes.